lib/grpnames.gi

```diff
@@ -20,46 +20,575 @@
 ##  from GAP3 code written by Markus Püschel and Sebastian Egner.
 ##
 
+
+#############################################################################
+##
+#M  IsTrivialNormalIntersectionInList( <L>, <U>, <V> ) . . . . generic method
+##
+InstallMethod( IsTrivialNormalIntersectionInList,
+               "if minimal normal subgroups are computed",
+               [ IsList, IsGroup, IsGroup ],
+
+  function( L, U, V )
+    local gs;
+
+    gs:= List(Filtered(L, N -> not IsTrivial(N)),
+                                               N -> GeneratorsOfGroup(N)[1]);
+    return ForAll(gs, g -> not g in U or not g in V);
+  end);
+
+#############################################################################
+##
+#M  IsTrivialNormalIntersection( <G>, <U>, <V> ) . . . . . . . generic method
+##
+
+InstallMethod( IsTrivialNormalIntersection,
+               "if minimal normal subgroups are computed",
+               [ IsGroup and HasMinimalNormalSubgroups, IsGroup, IsGroup ],
+
+  function( G, U, V )
+    local gs;
+
+    gs := List(MinimalNormalSubgroups(G), N -> GeneratorsOfGroup(N)[1]);
+    return ForAll(gs, g -> not g in U or not g in V);
```

Comment:
I doubt this is correct, as in principle, it would be legal for that generator to be the identity element (this is not what GAP currently *does*, but that's an artifact of the implementation). 
You probably want to look at the first non-trivial generator. And add a comment why that is a valid optimization.